src/SumoCoders/FrameworkCoreBundle/Tests/Mail/MessageFactoryTest.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace SumoCoders\FrameworkCoreBundle\Tests\Mail;
+
+use SumoCoders\FrameworkCoreBundle\Mail\MessageFactory;
+
+class MessageFactoryTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var MessageFactory
+     */
+    protected $messageFactory;
+
+    protected function setUp()
+    {
+        $temp = tempnam(sys_get_temp_dir(), 'message_factory_test');
+        $this->messageFactory = new MessageFactory(
+            $this->getTemplating(),
+            'mails/default_email.html.twig',
+            $temp
+        );
+    }
+
+    /**
+     * @inherit
+     */
+    protected function tearDown()
+    {
+        $this->messageFactory = null;
+    }
+
+    /**
+     * @return \PHPUnit_Framework_MockObject_MockObject
+     */
+    protected function getTemplating()
```

Comment:
I'd call this "getTemplateingMock".

But why don't you use a real Twig instance here? This would enable you to test the creation of html emails. Your html test now does not test anything about the rendering.